test/js/tests/ResourcesController.test.ts

```diff
@@ -115,6 +124,11 @@ describe('ResourcesController RESTful routes', () => {
       expect(edit.get({ id: '123' }).url).toBe('/resources/123/edit')
       expect(edit.get('456').url).toBe('/resources/456/edit')
     })
+
+    it('throws an error when not passing the id', () => {
+      // @ts-expect-error - Testing error handling
+      expect(() => edit.url().path).toThrow()
+    })
```

Comment:
I kept seeing Typescript errors on the generated code, so to improve this, I moved out of Elixir the core logic of URL generation. This way, now the CI can check it. Also added a lot of unit tests to this function. I feel more confident now in the correctness of the output